qa/rpc-tests/mempool_hard_fork_cleaning.py

```diff
@@ -0,0 +1,128 @@
+        # Fork 8 - Sidechain hard fork
+        # ####################################################################################################
+
+        mark_logs("Node 0 starts mining to reach the block immediately before the sidechain hard fork activation", self.nodes, DEBUG_MODE)
+        sc_fork_height = ForkHeights['MINIMAL_SC']
+        blocks = self.nodes[0].generate(sc_fork_height - 1) # The mempool should accept sidechain txs one block before the hard fork
+        self.sync_all()
+
+        mark_logs("Node 0 creates a v0 sidechain", self.nodes, DEBUG_MODE)
+        test_helper = BlockchainHelper(self)
+        test_helper.create_sidechain("v0_sidechain", SidechainParameters["DEFAULT_SC_V0"], EXPECT_SUCCESS)
+        self.sync_all()
+
+        mark_logs("Check that the nodes discard the sidechain transaction after disconnecting one block", self.nodes, DEBUG_MODE)
+        for node in self.nodes:
+            assert_equal(len(node.getrawmempool()), 1)              # All nodes should have the sidechain creation transaction
+            node.invalidateblock(blocks[-1])                        # Let's disconnect the last block
+            assert_equal(node.getblockcount(), sc_fork_height - 2)  # At this point, the sidechain transaction can't be mined anymore
+            assert_equal(len(node.getrawmempool()), 0)              # Nodes should have evicted the sidechain creation transaction
+
+        mark_logs("RemoveStaleTransaction() function worked properly for block disconnection on a hard fork", self.nodes, DEBUG_MODE)
+
+
+
+
+
+        # ####################################################################################################
+        # Fork 11 - Shielded pool deprecation hard fork
+        # ####################################################################################################
+
+        mark_logs("Node 0 starts mining to get close to the Shieleded Pool Deprecation hard fork [Fork - 3]", self.nodes, DEBUG_MODE)
+        shielded_pool_deprecation_fork_height = ForkHeights['SHIELDED_POOL_DEPRECATION']
+        block_count = self.nodes[0].getblockcount()
+        self.nodes[1].generate(shielded_pool_deprecation_fork_height - block_count - 3)
+        self.sync_all()
+
+        mark_logs("Split the network", self.nodes, DEBUG_MODE)
+        self.split_network(0)
+
+        mark_logs("Node 0 creates a shielding transaction", self.nodes, DEBUG_MODE)
+        total_funds_before_shielding = sum(x['amount'] for x in self.nodes[0].listunspent(0))
+        opid = self.nodes[0].z_mergetoaddress(["ANY_TADDR"], self.nodes[0].z_getnewaddress(), Z_FEE, 2, 2)["opid"]
+        wait_and_assert_operationid_status(self.nodes[0], opid)
+
+        # Give time to the wallet to be notified and updated the balance (listunspent would otherwise return an outdated value)
+        wait_until(self.nodes[0].getmempoolinfo()['fullyNotified'] == True, 20)
```

Comment:
`wait_until()` takes a callable argument, but the current one - `self.nodes[0].getmempoolinfo()['fullyNotified'] == True` - is not, hence it generates a few warnings like these:
`Warning: exception "'bool' object is not callable" occured in test_framework.util`

I suggest to change that to:
`wait_until(lambda: self.nodes[0].getmempoolinfo()['fullyNotified'] == True, 20)`

Reply:
Nice catch, thanks! My original version wasn't even working as expected, it was in practice like a `time.sleep()`, now it's much better!